### ISSUE TITLE
Add a filter syntax validator

### ISSUE THREAD

**Issue by curquiza** [meilisearch/documentation#1255]
I know this is not the priority and it involves tech people working on this, but it should be definitely a great addition, so I don't want we forget it 😇 

We could have a filter syntax validator as [Algolia did](https://www.algolia.com/doc/api-reference/api-parameters/filters/#filters-syntax-validator).

I ping @irevoire who already did a playground to test `filter` 😄

**Comment by bidoubiwa** [comment]
If we want a filter syntax validator it is different than just using the response answers from MeiliSearch.
MeiliSearch also returns errors based on the absence of a certain field in the `filterableAttributes` that would not be relevant in the case of a syntax validator. 
So it might be tricky, unless we make a playground on a certain dataset

**Comment by curquiza** [comment]
Yes, the playground of tamo is not the same as the syntax validator, it's true. Sorry for the ping tamo!
However, a syntax validator is still a good idea I think 😇 but not a priority of course!

**Comment by gmourier** [comment]
Wouldn't it be more relevant to have this in the mini-dashboard where it allows to check on real data and display the results that match the filters?

**Comment by curquiza** [comment]
I don't know if the mini dashboard is planned to be improved because of the SaaS

**Comment by bidoubiwa** [comment]
> But to integrate a simple syntax validator in the documentation I would need to export something else, like convert my crate to wasm and make it a npm library or something like that 

🤩

**Comment by guimachiavelli** [comment]
After reading the discussion, I have the impression there are two different understandings of what a filter syntax validator should be.

1. What Algolia has on their website seems to be a naïve syntax checker. It doesn't seem to be connected to an Algolia instance and does not return errors for e.g. attributes not present in the `filterableAttributes` list.
2. A more complex validator reliant on Meilisearch errors and thus somehow connected to a Meilisearch instance (possibly via wasm).

Option 2 seems extremely complex and time consuming. I might be wrong, but it seems like too much energy for little benefit to users.

Option 1 seems feasible, though I have some concerns regarding keeping the validator up to date when the de facto maintainers are an external team.

I will pursue option 1 with basement. To avoid splitting the discussion in two places, I will close this issue and track it in their preferred tool. I will keep you updated about progress on this via the monthly docs website report.
